jmeter/src/main/java/com/excelian/mache/jmeter/mongo/MongoConsoleMultiThreadedTester.java

```diff
@@ -0,0 +1,286 @@
+	    }
+	}
+
+	static class WriterCallable extends MongoBaseCallable implements
+			Callable<Statistics> {
+		private static AtomicReference<String> lastWrittenValue = new AtomicReference<String>(
+				null);
+		private final int writerSleepMs;
+		private final String entityKey;
+
+		public WriterCallable(int writerSleepMs, String entityKey,
+				final String mongoIP, final int mongoPort,
+				final String activeMq, final String mongoKeyspace)
+				throws JMSException {
+			super(mongoIP, mongoPort, activeMq, mongoKeyspace);
+			this.writerSleepMs = writerSleepMs;
+			this.entityKey = entityKey;
+
+		}
+
+		public static String getLastWrittenValue() {
+			return lastWrittenValue.get();
+		}
+
+		@Override
+		public Statistics call() throws InterruptedException {
+			final Statistics result = new Statistics();
+			result.milisStarted = System.currentTimeMillis();
+
+			for (int i = 0; i < 20000000; ++i) {
+				if (!Thread.interrupted()) {
+					String nValue = generateValue();
+					MongoTestEntity tE = new MongoTestEntity(this.entityKey,
+							nValue);
+					//cache1.put(this.entityKey, tE);
+					lastWrittenValue.set(nValue);
+					result.runsCount++;
+					result.okCount++;
+					Thread.sleep(writerSleepMs);
+				}
+			}
+
+			result.milisEnded = System.currentTimeMillis();
+			
+			teardown();
+			return result;
+		}
+
+		private String generateValue() {
+			return Generators.timeBasedGenerator().generate().toString();
+		}
+	}
+
+	static class ReaderCallable extends MongoBaseCallable implements Callable<Statistics> {
+		private final int readerSleepMs;
+		private final int readerTimeoutMs;
+		private final String entityKey;
+
+		public ReaderCallable(int readerSleepMs, int readerTimeoutMs,
+				String entityKey, final String mongoIP, final int mongoPort,
+				final String activeMq, final String mongoKeyspace)
+				throws JMSException {
+			super(mongoIP, mongoPort, activeMq, mongoKeyspace);
+			this.readerSleepMs = readerSleepMs;
+			this.readerTimeoutMs = readerTimeoutMs;
+			this.entityKey = entityKey;
+		}
+
+
+		@Override
+		public Statistics call() throws InterruptedException {
+			while(WriterCallable.getLastWrittenValue() == null) {
+				Thread.sleep(0);
+			}
+			
+			final Statistics result = new Statistics();
+			result.milisStarted = System.currentTimeMillis();
+
+			for (int i = 0; i < 500000; ++i) {
+				if (!Thread.interrupted()) {
+					long started = System.currentTimeMillis();
+					String nValue = null;
+					do {
+						//nValue = cache1.get(this.entityKey).description;
+						if (WriterCallable.getLastWrittenValue().equals(nValue)) {
+							break;
+						}
+						Thread.sleep(readerSleepMs);
+					} while (!WriterCallable.getLastWrittenValue().equals(nValue) || System.currentTimeMillis() - started < readerTimeoutMs);
+					result.runsCount++;
+					if (WriterCallable.getLastWrittenValue().equals(nValue))
+						result.okCount++;
+					else
+						result.errorsCount++;
+				}
```

Comment:
not sure about this, I've been using `CountdownLatch.wait(timeout)` for some stuff similar to this.